src/Types/UserGroup.elm

```diff
@@ -0,0 +1,285 @@
+        , ( "company_name", address.companyName |> Maybe.map Encode.string |> Maybe.withDefault Encode.null )
+        , ( "address", address.address |> Maybe.map Encode.string |> Maybe.withDefault Encode.null )
+        , ( "zip", address.zip |> Maybe.map Encode.string |> Maybe.withDefault Encode.null )
+        , ( "city", address.city |> Maybe.map Encode.string |> Maybe.withDefault Encode.null )
+        , ( "country", address.country |> Maybe.map Encode.string |> Maybe.withDefault Encode.null )
+        ]
+
+
+decodeContactDetails : Decode.Decoder ContactDetailsDto
+decodeContactDetails =
+    Decode.map2 ContactDetailsDto
+        (Decode.field "inherited_from" (Decode.maybe decodeUserGroupId))
+        (Decode.field "address" decodeAddress)
+
+
+encodeContactDetails : ContactDetailsDto -> Encode.Value
+encodeContactDetails contactDetails =
+    Encode.object
+        [ ( "inherited_from", contactDetails.inheritedFrom |> Maybe.map encodeUserGroupId |> Maybe.withDefault Encode.null )
+        , ( "address", encodeAddress contactDetails.address )
+        ]
+
+
+decodeTag : Decode.Decoder TagDto
+decodeTag =
+    Decode.map2 TagDto
+        (Decode.field "name" Decode.string)
+        (Decode.oneOf
+            [ Decode.field "value" (Decode.maybe Decode.string)
+            , Decode.succeed Nothing
+            ]
+        )
+
+
+encodeTag : TagDto -> Encode.Value
+encodeTag tag =
+    case tag.value of
+        Just value ->
+            Encode.object
+                [ ( "name", Encode.string tag.name )
+                , ( "value", Encode.string value )
+                ]
+
+        Nothing ->
+            Encode.object
+                [ ( "name", Encode.string tag.name )
+                ]
+
+
+decodeUserGroupDto : Decode.Decoder UserGroupDto
+decodeUserGroupDto =
+    Decode.map7 UserGroupDto
+        (Decode.field "id" decodeUserGroupId)
+        (Decode.field "parent_id" (Decode.maybe decodeUserGroupId))
+        (Decode.field "name" Decode.string)
+        (Decode.field "children" (Decode.list decodeChildUserGroup))
+        (Decode.field "settings" decodeSettings)
+        (Decode.field "contact_details" decodeContactDetails)
+        (Decode.field "tags" (Decode.list decodeTag))
+
+
+encodeUserGroupDto : UserGroupDto -> Encode.Value
+encodeUserGroupDto userGroup =
+    Encode.object
+        [ ( "id", encodeUserGroupId userGroup.id )
+        , ( "parent_id", userGroup.parentId |> Maybe.map encodeUserGroupId |> Maybe.withDefault Encode.null )
+        , ( "name", Encode.string userGroup.name )
+        , ( "children", Encode.list encodeChildUserGroup userGroup.children )
+        , ( "settings", encodeSettings userGroup.settings )
+        , ( "contact_details", encodeContactDetails userGroup.contactDetails )
+        , ( "tags", Encode.list encodeTag userGroup.tags )
+        ]
+
+
+parseInt : String -> Decode.Decoder Int
+parseInt str =
+    case String.toInt str of
```

Comment:
Q: How will you deal with the really long numbers that doesn't accommodate in Int type? E.g. `"97642954568123545688"` is a valid ID for us.